fluss-lake/fluss-lake-iceberg/src/test/java/com/apache/fluss/lake/iceberg/catalog/IcebergLakeCatalogTest.java

```diff
@@ -0,0 +1,334 @@
+                        .column("order_id", DataTypes.INT())
+                        .column("customer_name", DataTypes.STRING())
+                        .column("created_at", DataTypes.TIMESTAMP_LTZ()) // timestamp with zone
+                        .column("processed_at", DataTypes.TIMESTAMP()) // timestamp without zone
+                        .primaryKey("order_id")
+                        .build();
+
+        TableDescriptor tableDescriptor =
+                TableDescriptor.builder().schema(flussSchema).distributedBy(4, "order_id").build();
+
+        org.apache.iceberg.Schema icebergSchema =
+                flussIcebergCatalog.convertToIcebergSchema(tableDescriptor);
+
+        assertThat(icebergSchema.findField("order_id")).isNotNull();
+        assertThat(icebergSchema.findField("customer_name")).isNotNull();
+        assertThat(icebergSchema.findField("created_at")).isNotNull();
+        assertThat(icebergSchema.findField("processed_at")).isNotNull();
+
+        assertThat(icebergSchema.findField("created_at").type())
+                .isEqualTo(Types.TimestampType.withZone());
+        assertThat(icebergSchema.findField("processed_at").type())
+                .isEqualTo(Types.TimestampType.withoutZone());
+
+        assertThat(icebergSchema.findField(BUCKET_COLUMN_NAME)).isNotNull();
+        assertThat(icebergSchema.findField(OFFSET_COLUMN_NAME)).isNotNull();
+        assertThat(icebergSchema.findField(TIMESTAMP_COLUMN_NAME)).isNotNull();
+
+        assertThat(icebergSchema.columns()).hasSize(7);
+    }
+
+    @Test
+    void createsPrimaryKeyTableBucketPartitionedOnly() {
+        String database = "test_db";
+        String tableName = "pk_table";
+
+        createNamespaceIfAbsent(database);
+
+        // primary key table
+        Schema flussSchema =
+                Schema.newBuilder()
+                        .column("shop_id", DataTypes.BIGINT())
+                        .column("user_id", DataTypes.BIGINT())
+                        .column("num_orders", DataTypes.INT())
+                        .column("total_amount", DataTypes.INT())
+                        .primaryKey("user_id")
+                        .build();
+
+        TableDescriptor tableDescriptor =
+                TableDescriptor.builder().schema(flussSchema).distributedBy(4, "user_id").build();
+
+        TableIdentifier tableId = TableIdentifier.of(database, tableName);
+
+        org.apache.iceberg.Schema icebergSchema =
+                flussIcebergCatalog.convertToIcebergSchema(tableDescriptor);
+
+        PartitionSpec expectedPartitionSpec =
+                flussIcebergCatalog.createPartitionSpec(tableDescriptor, icebergSchema);
+
+        SortOrder expectedSortOrder =
+                SortOrder.builderFor(icebergSchema).asc(OFFSET_COLUMN_NAME).build();
+
+        Map<String, String> tableProperties = new HashMap<>();
+        tableProperties.put("write.merge.mode", "merge-on-read");
+
+        Table createdTable =
+                icebergCatalog
```

Comment:
we don't care about iceberg's catalog can create table or not since it's iceberg implementation. So, no need to verify the table create via Iceberg catalog provided by iceberg. We need to verify the table create via Fluss `IcebergLakeCatalog`